### DISCUSSION TITLE
Fast DB vs ShivamShrirao DB

### DISCUSSION THREAD

**Discussion by toyxyz** [TheLastBen/fast-stable-diffusion#283] [General]
I trained the following character images using Fast DB and ShivamShrirao DB and compared the results.

https://github.com/ShivamShrirao/diffusers/tree/main/examples/dreambooth

![unknown (1)](https://user-images.githubusercontent.com/8006000/198823366-9fe1de3f-4085-4ad7-ad81-76c064296e35.png)

Both trained at 1500 steps and took similarly about 20 minutes.

The result is as follows. 

Euler, 28steps, CFG 11, Clipskip 1

prompt : ba-shiroko, masterpiece, best quality

negative : nsfw, lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry

The four on the left are Shiv, and the four on the right are Fast DB. Bikes and gloves included in the imageset appear stronger in Shiv.

![20221029180727084](https://user-images.githubusercontent.com/8006000/198823400-e3bb974b-99bf-4c6f-9c73-c40360953c6e.jpg)

![tmpflxjqz44](https://user-images.githubusercontent.com/8006000/198823969-e3acfc4e-1753-428a-a8b4-e2be672adb91.png)

The Regularization-Images I used with shiv are this.

![unknown (2)](https://user-images.githubusercontent.com/8006000/198823662-7a8a0c40-928d-4097-a5b8-46cc47ad9805.png)

**Comment by TheLastBen** [comment]
you named your images something like 0000-0-a 0.1.jpg, with the fast method, the instance name/prompt would become "--a" which is a bad instance name, it interferes with the letter "a".

And you didn't summon my method in your prompt because you didn't use the instance name.

The results you got from my method are not actually the result of the training.

Try again the NEW method by renaming the instance pictures to "jjgkkgjgkl (1) .... etc" and in the A1111, call the method by using "**jjgkkgjgkl**" instead of "**ba-shiroko**".

**Reply by toyxyz** [reply]
I forgot to mention, the image attached here was captured before I renamed the file in the middle of the job. File name was changed like this.

<img width="244" alt="20221029184529457" src="https://user-images.githubusercontent.com/8006000/198824849-463cc961-1370-445f-946b-d8e6e14c70c9.png">

Is there any need to avoid using '-' or spaces? like ba-shiroko (1) -> bashiroko (1)

**Reply by TheLastBen** [reply]
ba-shiroko will interfere with the previous model knowledge, you're reducing the quality by half, you need to use a password-like instance name that also doesn't contain known abbreviations, like the example before : jjgkkgjgkl (1).jpg

**Reply by toyxyz** [reply]
I trained using jjgkkgjgkl as per your advice. (1500 steps)

Euler, 28steps, CFG 11, Clipskip 1

prompt : jjgkkgjgkl, masterpiece, best quality

negative : nsfw, lowres, bad anatomy, bad hands, text, error, missing fingers, extra digit, fewer digits, cropped, worst quality, low quality, normal quality, jpeg artifacts, signature, watermark, username, blurry

1. Fast DB + jjgkkgjgkl /    2. Fast DB + ba-shiroko /   3. Shiv + ba-shiroko 
 
![20221029201406574](https://user-images.githubusercontent.com/8006000/198828287-03013a9c-db70-48bc-b3ad-1eba6f65b56c.jpg)


Likewise, Shiv seems to have a bit more character. By the way when I checked the learning rate and batch size, the default settings were as follows. Could this be the cause of the difference?

shiv

--learning_rate=1e-6

  --sample_batch_size=4

fast-db

--train_batch_size=1 \

--learning_rate=2e-6 \

**Reply by TheLastBen** [reply]
didn't you notice the overly incorrect anatomy with Shiv's method ?

also, try a real person so we can distinguish between the methods more efficiently

**Reply by toyxyz** [reply]
I picked a random influencer that wasn't famous from Google and trained it on Fast DB and Shiv. (Step :1800) The training time is about the same, about 20-30 minutes. Confirm that this person's ID is not in SD 1.5. She always seems to have a habit of avoiding taking pictures of the front of her face. This time Fast-DB seems to have been trained a bit better.

Image set used for training
![20221030084250006](https://user-images.githubusercontent.com/8006000/198857098-8834736c-7112-4ea3-9fc1-dece2cb9f728.jpg)

Euler A, 70 steps, CFG 7

prompt : a photo of efxdrd at studio, direct picture, beautiful face

negative : ((((ugly)))), (((duplicate))), ((morbid)), ((mutilated)), [out of frame], extra fingers, mutated hands, ((poorly drawn hands)), ((poorly drawn face)), (((mutation))), (((deformed))), ((ugly)), blurry, ((bad anatomy)), (((bad proportions))), ((extra limbs)), cloned face, (((disfigured))), out of frame, ugly, extra limbs, (bad anatomy), gross proportions, (malformed limbs), ((missing arms)), ((missing legs)), (((extra arms))), (((extra legs))), mutated hands, (fused fingers), (too many fingers), (((long neck))), nsfw

1. Shiv 1800 steps / 2. Shiv 800 steps / 3. Fast DB 1800 steps

![a_xy_grid-0389-a_photo_of_efxdrd_at_studio,_direct_picture,_beautiful_face](https://user-images.githubusercontent.com/8006000/198857156-66c195a1-ef3b-48a2-ad52-0e7e7aae4553.jpg)
![a_xy_grid-0393-a_photo_of_efxdrd_at_studio,_direct_picture](https://user-images.githubusercontent.com/8006000/198857157-39459546-4d06-44d1-ba2c-d3a7ae633afd.jpg)

**Reply by TheLastBen** [reply]
3000 steps is the optimal steps count for the new method

**Reply by SteinarVatne** [reply]
I trained a network with ~120 images of myself and 5 collegues (varies between 15 to 30 instance images pr. person) with the fast method. I used 10000 steps. For image/instance names, I used "FirstnameLastname" without spaces, like I've seen others do. The basic likenesses of the subjects are very good, and it also seems like the network preserves existing concepts better (inference with just "photo of person" on older versions would give poor results). However, this fast model seems to have other issues; I find it more difficult to mix or finetune these concepts with other existing concepts, and when using img2img or inpainting, the trained concepts tend to introduce a lot of artifacts/blotches in the generated images?

**Reply by TheLastBen** [reply]
Use the weights on the styles ((example)) to apply them easily on the trained subjects, and reduce the cfg scale to 7 or 6.5

**Reply by SteinarVatne** [reply]
I think maybe I'm doing the opposite of what you describe; I'm applying the trained concept to an existing image. However, I tried applying the trained concept with a reduced weight, using [LastnameFirstname:0.1] for example, and although the blotchiness is perhaps a little bit better, it's still there. Actually, there is blotchiness even when I try this with existing concepts, like "johnny depp", artifacts that are not present if I switch to original weights, or weights with concepts trained on the older method, witch suggest that the whole network has somehow deteriorated. This blotchiness appears similar to descriptions I've seen of overfitting, perhaps I should try with fewer steps?

**Reply by TheLastBen** [reply]
In a few hours I will add a feature to train the text_encoder only for a certain amount of steps, it will fix the issue with the style transfer, but you'll need a bit more steps

**Reply by toyxyz** [reply]
stabilityai/sd-vae-ft-mse is said to help create a photo of a realistic person, but adversely affects the creation of a 2D animated character. (blurry) Does Fasf-DB use this vae for training? If so, it would be nice to have an option for the user to decide whether to apply or not.

**Reply by TheLastBen** [reply]
I'll add it as an option in the "download model cell", but there need to be a comparison to make sure it actually affects the quality

**Reply by nawnie** [reply]
i can run a test tonight, my current test was going to be to add more names, such as naming a file Goku_face  (1), Goku_Shoe_(1), Goku_full body (1), not seeing much documentation on how this new setup works (in detail) maybe im missing something just found this today. or would my better bet be to add some lines to accelerate to include a concept list or identifier. i love the option to use the text encoder is there anyway to see what it is labeling the images (im guessing) its like the files used in embedding from 111 UI. if so id probably like to change them for better results (pictures look good im training 30 objects at once so... havnt done much testing on end result yet.
